aspnet/fundamentals/static-files.rst

```diff
@@ -185,110 +144,60 @@ As with the ``UseStaticFiles``, ``UseDefaultFiles``, and ``UseDirectoryBrowser``
+The ASP.NET static file middleware understands almost 400 known file content types. If the user requests a file of an unknown file type, the static file middleware returns a HTTP 404 (Not found) response. If directory browsing is enabled, a link to the file will be displayed, but the URI will return an HTTP 404 error.
 
-  public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
-  {
-    ...
+The following code enables serving unknown types and will render the unknown file as an image. See  this `list of common MIME content types <http://www.iana.org/assignments/media-types/media-types.xhtml>`__.
 
-    // Allow directory browsing.
-    app.UseDirectoryBrowser();
+.. literalinclude:: static-files/sample/StartupUnKnown.cs
+  :language: c#
+  :lines: 55-66
+  :dedent: 12
 
-    // Set up custom content types - associating file extension to MIME type
-    var provider = new FileExtensionContentTypeProvider();
-    provider.Mappings.Add(".myapp", "application/x-msdownload");
+With the code above, a request for a file with an unknown content type will be returned as an image.
 
-    // Serve static files.
-    app.UseStaticFiles(new StaticFileOptions { ContentTypeProvider = provider });
+.. warning:: Enabling :dn:property:`~Microsoft.AspNetCore.Builder.StaticFileOptions.ServeUnknownFileTypes` is a security risk and using it is discouraged.  :dn:class:`~Microsoft.AspNetCore.StaticFiles.FileExtensionContentTypeProvider`  (explained below) provides a safer alternative to serving files with non-standard extensions.
 
-    ...
+FileExtensionContentTypeProvider
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Now, if the user attempts to browse to any file with an extension of ``.myapp``, the user will be prompted to download the file (or it will happen automatically depending on the browser).
+The :dn:class:`~Microsoft.AspNetCore.StaticFiles.FileExtensionContentTypeProvider` class contains a  collection that maps file extensions to MIME content types. In the following sample, several file extensions are registered to known MIME types.
 
-IIS Considerations
-------------------
+.. literalinclude:: static-files/sample/StartupFECTP.cs
+  :language: c#
+  :lines: 58-70
+  :dedent: 12
 
-ASP.NET Core applications hosted in IIS use the HTTP platform handler to forward all requests to the application including requests for static files. The IIS static file handler is not used because it won’t get a chance to handle the request before it is handled by the HTTP platform handler.
 
-Best practices
---------------
+Considerations
+^^^^^^^^^^^^^^^^
 
-This section includes a list of best practices for working with static files:
+- ASP.NET Core applications hosted in IIS use the ASP.NET Core Module to forward all requests to the application including requests for static files. The IIS static file handler is not used because it doesn't get a chance to handle requests before they are handled by the ASP.NET Core Module.
```

Comment:
@blowdart  how's this

- To remove the IIS static file handler (at the server or or website level):

    - Navigate to the **Modules** feature
    - Select **StaticFileModule** in the list
    - Tap **Remove** in the **Actions** sidebar
    
.. warning:: If the IIS static file handler is enabled **and** the ASP.NET Core Module (ANCM) is not correctly configured (for example if *web.config* was not deployed), static files will be served.